server/pulp/server/db/model/__init__.py

```diff
@@ -553,8 +553,14 @@ def save_with_set_on_insert(self, fields_to_set_on_insert):
         set_on_insert = {}
         for field in fields_to_set_on_insert:
             set_on_insert[field] = stuff_to_update.pop(field)
+
         task_id = stuff_to_update.pop('task_id')
 
+        # Don't try to set Mongoengine internal values.
+        # Can cause silent failures if you accidentally overwrite 'id'.
+        for internal_field in ('id', '_ns'):
```

Comment:
How did you choose which fields are internal? E.g. `_id` is not in the tuple.

Reply:
@goosemania Mongoengine uses `id` as an alias for `_id`

Reply:
@asmacdo , thank you

Reply:
@goosemania exactly, and as far as methods go, I interactively debugged the _data object to determine what was being added to it.